udev.c

```diff
@@ -147,9 +147,15 @@ enum udev_status udev_wait_for_events(int seconds)
 	tv.tv_sec = seconds;
 	tv.tv_usec = 0;
 
-	if (select(fd + 1, &readfds, NULL, NULL, &tv) > 0 && FD_ISSET(fd, &readfds))
-		if (udev_monitor_receive_device(udev_monitor))
-			return UDEV_STATUS_SUCCESS; /* event detected */
+	if (select(fd + 1, &readfds, NULL, NULL, &tv) > 0 && FD_ISSET(fd, &readfds)) {
+		struct udev_device *dev = udev_monitor_receive_device(udev_monitor);
+
+		if (dev)
+			udev_device_unref(dev);
+		else
+			return UDEV_STATUS_ERROR;
+		return UDEV_STATUS_SUCCESS; /* event detected */
```

Comment:
Nitpick: IMO it would look better if the `return UDEV_STATUS_SUCCESS` was moved into the `if` clause.